### PR TITLE
 Edit task endpoint 

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
**Description of changes**
- Added Edit task endpoint to the backend.

**Associated ticket link**
https://trello.com/c/K0SN8e8t

**Screenshot / Screen Recording**

https://github.com/user-attachments/assets/c54e35c2-3c16-4d6e-9fdf-d4bc76e36b33

### CODE REVIEW

**Comment by ragib-hossain-sazim** [comment]
Just a suggestion.
Please highlight the things you have done in the PR when writing description.
for eg. 

- Added a new update-task-dto
- Added [PLACEHOLDER] methods inside tasks.service
- [High level explanation of any special code logic that the reviewer should be aware of (optional)]

File: src/tasks/tasks.repository.ts [hunk 1]
```diff
@@ -21,4 +21,10 @@ export class TasksRepository extends EntityRepository<Task> {
       );
     }
   }
+
+  async updateTask(task: Task, taskData: Partial<Task>): Promise<Task> {
```
Comment (ragib-hossain-sazim) [inline src/tasks/tasks.repository.ts:25]: [Nitpick] `updateTask` => `updateOne`

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending nit comments

File: src/tasks/dtos/update-task.dto.ts
Comment (ishmam-iqbal-sazim) [inline src/tasks/dtos/update-task.dto.ts:1]: Can put all the dtos in tasks.dtos.ts file inside the tasks folder. Probably best not to make separate dto files unless necessary.

File: src/tasks/tasks.service.ts [hunk 2]
```diff
@@ -22,4 +23,9 @@ export class TasksService {
   async findAllTasks(): Promise<Task[]> {
     return await this.tasksRepository.findAll();
   }
+
+  async updateTask(id: number, taskData: UpdateTaskDto): Promise<Task> {
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.service.ts:27]:
```suggestion
  async updateTask(id: number, updateTaskDto: UpdateTaskDto): Promise<Task> {
```

Let's name the param consistently with the type.

**Pull request merged**